### PR TITLE
End of game - print screens, ending conditions, exit action

### LINKED ISSUE #42: PrintWinScreen()

### LINKED ISSUE #43: PrintLooseScreen()

### LINKED ISSUE #59: Game::exit()

### PR DESCRIPTION
**Pull request by niemarcin**
closes #42 
closes #43 
closes #59

### CODE REVIEW

File: shm/Game.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@ enum class Action {
 class Game {
 private:
     size_t money_;
-    size_t days_;
+    const size_t days_;
```
Comment (J-Cieplinski) [inline shm/Game.hpp]:
```suggestion
    const size_t timeConstraint;
```

Comment (J-Cieplinski) [reply shm/Game.hpp]: Or similar. It shouldn't stays as `days_` since it's misleading

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

File: shm/Game.cpp [hunk 2]
```diff
@@ -4,15 +4,15 @@
 
 constexpr size_t distancePerDay{2};
 
-Game::Game(size_t money, size_t days, size_t final_goal)
-    : money_(money), days_(days), final_goal_(final_goal) {}
+Game::Game(size_t money, size_t days, size_t finalGoal)
```
Comment (J-Cieplinski) [inline shm/Game.cpp]:
```suggestion
Game::Game(size_t money, size_t timeLimit, size_t finalGoal)
```

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

**Review by J-Cieplinski — APPROVED** [review]

File: shm/Game.cpp [hunk 3]
```diff
@@ -4,18 +4,80 @@
...
+        "   ##    ##     ## ##          ##       ####  ## ##     ##" << "\n" <<  
+        "   ##    ######### ######      ######   ## ## ## ##     ##" << "\n" << 
+        "   ##    ##     ## ##          ##       ##  #### ##     ##" << "\n" << 
+        "   ##    ##     ## ##          ##       ##   ### ##     ##" << "\n" << 
+        "   ##    ##     ## ########    ######## ##    ## ######## " << "\n";
+}
+
+void Game::printWinScreen() const {
+    std::string horizontalSeparator(59, '=');
+    int i = 0;
```
Comment (iprice73) [inline shm/Game.cpp]: Does this line have a purpose, I do not see any use of it later in code

File: shm/Game.cpp [hunk 4]
```diff
@@ -4,18 +4,80 @@
...
+              << player_->getMoney() 
+              << "\n"
+              << std::string(20, ' ')
+              << "CONGRATULATIONS" << "\n"
+              << horizontalSeparator << "\n";
+}
+
+void Game::printLoseScreen() const {
+    std::string horizontalSeparator(59, '=');
+    int i = 0;
```
Comment (iprice73) [inline shm/Game.cpp]: same here

File: shm/Game.cpp [hunk 5]
```diff
@@ -4,18 +4,80 @@
...
+              << player_->getMoney() 
+              << "\n"
+              << std::string(25, ' ')
+              << "TRY AGAIN" << "\n"
+              << horizontalSeparator << "\n";
+}
+
+void Game::printEndGameScreen() const {
+    std::string horizontalSeparator(59, '=');
+    int i = 0;
```
Comment (iprice73) [inline shm/Game.cpp]: and here

**Review by iprice73 — APPROVED** [review]

**Pull request merged**